art/attacks/evasion/auto_projected_gradient_descent.py

```diff
@@ -115,11 +115,30 @@ def __init__(
                     if is_probability(estimator.predict(x=np.ones(shape=(1, *estimator.input_shape)))):
                         raise NotImplementedError("Cross-entropy loss is not implemented for probability output.")
 
-                    self._loss_object = tf.reduce_mean(
-                        tf.keras.losses.categorical_crossentropy(
-                            y_pred=estimator._output, y_true=estimator._labels_ph, from_logits=True
-                        )
-                    )
+                    # self._loss_object = tf.reduce_mean(
+                    #     tf.keras.losses.categorical_crossentropy(
+                    #         y_pred=estimator._output, y_true=estimator._labels_ph, from_logits=True
+                    #     )
+                    # )
+                    # modification for image-wise stepsize update
+                    class CrossEntropyLoss:
+                        def __init__(self, reduction="mean"):
+                            self.ce = tf.keras.losses.categorical_crossentropy(
```

Comment:
snake_case naming style:
```suggestion
                            self.ce_loss = tf.keras.losses.categorical_crossentropy(
```